src/EditorFeatures/CSharpTest/QuickInfo/SemanticQuickInfoSourceTests.cs

```diff
@@ -4135,5 +4135,32 @@ public class Methods2
             var expectedDescription = $"void Methods1.Do(string x)";
             VerifyWithReferenceWorker(markup, MainDescription(expectedDescription));
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.QuickInfo)]
+        [WorkItem(4868, "https://github.com/dotnet/roslyn/issues/4868")]
+        public void QuickInfoExceptions()
+        {
+            Test(@"
+using System;
+namespace MyNs
+{
+    class MyException1 : Exception { }
+    class MyException2 : Exception { }
+    class TestClass
+    {
+        /// <exception cref=""MyException1""></exception>
+        /// <exception cref=""T:MyNs.MyException2""></exception>
+        /// <exception cref=""System.Int32""></exception>
+        /// <exception cref=""double""></exception>
+        /// <exception cref=""Not_A_Class_And_Therefore_Not_Listed""></exception>
+        void M()
+        {
+            M$$();
+        }
+    }
+}
+",
+                Exceptions($"\r\n{WorkspacesResources.Exceptions}\r\n  MyException1\r\n  MyException2\r\n  double\r\n  int"));
+        }
```

Comment:
By “parse as a type”, what do you mean? The type didn’t exist in the compilation? If so, I can see that occurring pretty easy – the callee throws an XamlException but don’t have a reference to System.Xaml.dll. In this case, I would expect it to show up.

From: Brett Forsgren [mailto:notifications@github.com]
Sent: Monday, August 31, 2015 11:24 AM
To: dotnet/roslyn <roslyn@noreply.github.com>
Cc: David Kean <David.Kean@microsoft.com>
Subject: Re: [roslyn] list exceptions from the doc comments when showing quick info (#4874)


In src/EditorFeatures/CSharpTest/QuickInfo/SemanticQuickInfoSourceTests.cs<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fdotnet%2froslyn%2fpull%2f4874%23discussion_r38342291&data=01%7c01%7cDavid.Kean%40microsoft.com%7c91f365da6ee44a87b78e08d2b231653d%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=nTKMyVDU8NKiCoZv7jU1s2Qs2AVVVP6ruAHfk5BekPc%3d>:

> +    class TestClass

> +    {

> +        /// <exception cref=""MyException1""></exception>

> +        /// <exception cref=""T:MyNs.MyException2""></exception>

> +        /// <exception cref=""System.Int32""></exception>

> +        /// <exception cref=""double""></exception>

> +        /// <exception cref=""Not_A_Class_And_Therefore_Not_Listed""></exception>

> +        void M()

> +        {

> +            M$$();

> +        }

> +    }

> +}

> +",

> +                Exceptions($"\r\n{WorkspacesResources.Exceptions}\r\n  MyException1\r\n  MyException2\r\n  double\r\n  int"));

> +        }

VS 2013 simply displays the types as listed in the doc comment, e.g.,

MyException1

MyNs.MyException2,

System.Int32,

double

Not_A_Class_And_Therefore_Not_Listed

Having thought about the difficulties in dealing with sorting the fully-qualified names vs. sorting the simplified names (which could change with each new using), I'm leaning towards no sorting at all and simply using the declaration order, but still letting the symbol display service simplify names where possible, e.g.,

MyException1

MyException2

int

double

But there's still the issue of what to do with exceptions that didn't parse as a type. The doc comment author thought it important to include an exception type here, but why?

—
Reply to this email directly or view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fdotnet%2froslyn%2fpull%2f4874%2ffiles%23r38342291&data=01%7c01%7cDavid.Kean%40microsoft.com%7c91f365da6ee44a87b78e08d2b231653d%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=sEXBG7KWPIavVfDnUbleOBpskOT9OrPmx1Kxzhgwnhk%3d>.